hackerrank/1-week-prep/day-4/mock-test-4.py

```diff
@@ -0,0 +1,16 @@
+def truckTour(petrol_pumps):
+    for i in range(len(petrol_pumps)):
+        if can_complete_circle(petrol_pumps, i):
+            return i
+    return None
+
+def can_complete_circle(petrol_pumps, current, cummulated_petrol=0, visited_pumps=0):
+    num_of_petrol_pumps = len(petrol_pumps)
+    while visited_pumps < num_of_petrol_pumps:
+        petrol_pump = petrol_pumps[current]
+        if petrol_pump[0] + cummulated_petrol < petrol_pump[1]:
+            return False
+        cummulated_petrol = cummulated_petrol + petrol_pump[0] - petrol_pump[1]
```

Comment:
Unpacking `petrol_pump` to variables such as `gas_offered, distance_to_next` (or sth similar) will make the code easier to understand.